container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/shared_helpers/shared_helpers.go

```diff
@@ -248,7 +252,7 @@ func GetMatchingUserServiceObjectsAndKubernetesResources(
 				continue
 			}
 
-			if _, found := filters.Statuses[kubernetesService.GetStatus()]; !found {
+			if _, found := filters.Statuses[kubernetesService.GetContainer().GetStatus()]; !found {
```

Comment:
@laurentluce this may be a remote chance, but if `GetContainer` is `nil` then we'll have a segfault? here and it won't give us much information. Maybe we can add a nil check before?

Reply:
I made the change and then reverted it.  I don't think it is worse it for the amount of extra code.  I checked the code and the container attribute is always set on this path.